maven-plugin/build.gradle

```diff
@@ -51,4 +68,64 @@ publishing {
+    outputs.file(testProperties)
+
+    def workspaceStateProperties = file("$buildDir/maven-work/workspacestate.properties")
+    outputs.file(workspaceStateProperties)
+
+    doLast {
+        // Generate 'workspacestate.properties' file
+        java.util.Properties workspaceProp = new java.util.Properties();
+        getRootProject().getAllprojects().each { prj -> 
+            def ext = prj.getExtensions().findByType(PublishingExtension.class);
+            if(ext != null) {
+                ext.getPublications().each { pub -> 
+                    addProject(workspaceProp, prj)
+                }
+            }
+        }
+        workspaceProp.store(new java.io.FileOutputStream(workspaceStateProperties), "workspace state file")
+
+        // Generate 'test.properties' file
+        // See https://github.com/takari/takari-plugin-testing-project/blob/master/testproperties.md
+        java.util.Properties testProp = new java.util.Properties();
+
+        testProp.setProperty("project.groupId", getProject().getGroup().toString());
+        testProp.setProperty("project.artifactId", getProject().getConvention().getPlugin(BasePluginConvention.class).getArchivesBaseName());
+        testProp.setProperty("project.version", getProject().getVersion().toString());
+
+        testProp.setProperty("classpath", "" + jar.archiveFile.get().asFile + ":" + configurations.runtimeClasspath.getAsPath());
+
+        testProp.setProperty("localRepository", System.getProperty("user.home") + "/.m2/repository");
+        testProp.setProperty("repository.0", "<id>central</id><url>https://repo.maven.apache.org/maven2</url><releases><enabled>true</enabled></releases><snapshots><enabled>false</enabled></snapshots>");
+        testProp.setProperty("updateSnapshots", "false");
+        // testProp.setProperty("globalSettingsFile", null);
+        File userSettingsFile = new File(System.getProperty("user.home") + "/.m2/settings.xml")
+        if(userSettingsFile.exists()) {
+            testProp.setProperty("userSettingsFile", userSettingsFile.getAbsolutePath());
+        }
+        testProp.setProperty("workspaceResolver", project.configurations.takariWorkspace.asPath);
+        testProp.setProperty("workspaceStateProperties", workspaceStateProperties.getAbsolutePath());
+
+        testProp.store(new java.io.FileOutputStream(testProperties), "See https://github.com/takari/takari-plugin-testing-project/blob/master/testproperties.md")
+    }
+}
+
+def void addProject(java.util.Properties workspaceProp, Project p) {
+    workspaceProp.setProperty(p.getGroup().toString() + ":" + p.getConvention().getPlugin(BasePluginConvention.class).getArchivesBaseName() + ":jar::" + p.getVersion().toString(), new File(p.getBuildDir(), "/libs/${p.getConvention().getPlugin(BasePluginConvention.class).getArchivesBaseName()}-${p.getVersion().toString()}.jar").getAbsolutePath()); // jar.archiveFile.get().asFile.getAbsolutePath() ??
```

Comment:
do we want to remove the comment at the end of the line?

Reply:
```suggestion
    workspaceProp.setProperty(p.getGroup().toString() + ":" + p.getConvention().getPlugin(BasePluginConvention.class).getArchivesBaseName() + ":jar::" + p.getVersion().toString(), new File(p.getBuildDir(), "/libs/${p.getConvention().getPlugin(BasePluginConvention.class).getArchivesBaseName()}-${p.getVersion().toString()}.jar").getAbsolutePath());
```